### DISCUSSION TITLE
Digital kamstrup, two meters how?

### DISCUSSION THREAD

**Discussion by Zorkny** [jomjol/AI-on-the-edge-device#1723] [Q&A]
Hi,

I have a kamstrup with digital digits, showing total m3 and also litre/hour.
I can get the total m3 to work fine, but the litre/hour is always error since it goes from high numbers to zero and back again it gets disregarded. I want the previous to be considered for the m3 but not for the litre/hour but it seems it doesnt work ok for me.

What is the recommended config for this scenario?

Thanks!

**Comment by jomjol** [comment]
Can you send a picture of your meter?

**Comment by Zorkny** [comment]
Its Kamstrup FlowIQ2200, attached.

![kamstrup_flowiq_2200](https://user-images.githubusercontent.com/23185319/210015286-f164fff3-5bee-437d-b418-64ac91f84042.png)

**Reply by caco3** [reply]
This project is designed for increasing numbers (like your m3).
If you want it for numbers which can decrease, you need to disable several checks in the settings.

However I wonder why you need your l/h rate. The project can provide you this as well through MQTT or REST!

**Reply by Zorkny** [reply]
I agree, but m3 will change a lot less than l/h, I think of l/h as decimals to m3 almost. 1 litre per hour is not much and reading that number can give very very good resolution and show water leaks or other things.

Still, thanks. Still need modify the camera a bit for focus and light, it gets misaligned all the time.

**Comment by caco3** [comment]
> I agree, but m3 will change a lot less than l/h, I think of l/h as decimals to m3 almost

So then set `DecimalShift` to `3` and all is handled in `l`.

**Reply by caco3** [reply]
I just updated the documentation to make it more clear, see https://github.com/jomjol/AI-on-the-edge-device/pull/1729/files

And something else, in case you use Homeassistant:
The new (and really nice) Energy Dashboard expects the value to be in `m3` or `gallons`:
![grafik](https://user-images.githubusercontent.com/1783586/210049402-b5deadc6-7d98-4a04-a096-4e3f10ebcccb.png)

So this seems to be kind of an industry standard and I strongly would suggest to stick with that!
As for the Energy Dashboard the values automatically get converted to `l` as, like you, most people prefer it.

**Reply by Zorkny** [reply]
I agree, but you miss the point. The secondary number shows l/h and this means it will react and change to "1" a LOT faster than the m3 meter will move by +1. (l/h is dm3/h meaning it will take 1000 times longer to find out that water is running with m3 than dm3). Or do I misunderstand something here?

**Reply by DanielWe2** [reply]
It looks like you have three decimal digits for your m3 number. 0.001 m3 is one liter. So your normal number already counts every single liter. I don't see how an additional liter per hour number can improve anything. (it is also unclear how the meter is calculating that).

**Reply by DanielWe2** [reply]
> 
> And something else, in case you use Homeassistant: The new (and really nice) Energy Dashboard expects the value to be in `m3` or `gallons`: ![grafik](https://user-images.githubusercontent.com/1783586/210049402-b5deadc6-7d98-4a04-a096-4e3f10ebcccb.png)
> 

That is not true anymore since the December release. It will handle meters in the unit liters if the device class is water. (the text in the config ui of your Screenshot doesn't reflect that change though).

**Reply by Zorkny** [reply]
@DanielWe2  Thanks! I had actually not noticed the comma and the decimals... this makes more sense now. Thanks for great support!

**Reply by caco3** [reply]
> That is not true anymore since the December release. It will handle meters in the unit liters if the device class is water. (the text in the config ui of your Screenshot doesn't reflect that change though).

Ah, good to know, I still am on the previous release.
